starter-code/app.js

```diff
@@ -1,5 +1,78 @@
+'use strict';
+
+const bodyParser = require('body-parser');
+const expressLayouts = require('express-ejs-layouts');
 const express = require('express');
 const app = express();
-const Chuck  = require('chucknorris-io');
+const Chuck = require('chucknorris-io');
 const client = new Chuck();
+// ...
+
+
+// ...
+
+app.use(express.static('public'));
+/* New */
+app.use(expressLayouts);
+app.set('layout', 'layouts/main-layout');
+app.set('views', __dirname + '/views');
+app.set('view engine', 'ejs');
+
+
+// ...
+
+// ...
+app.use(bodyParser.urlencoded({
+  extended: true
+}));
+
+
+
+app.get('/', (req, res, next) => {
+  res.send('hello world');
+});
+
+app.get('/random', (req, res, next) => {
+  // Retrieve a random chuck joke
+  client.getRandomJoke()
+    .then((joke) => {
+      res.send(joke.value);
+    }).catch((err) => {
+      console.log('Oh oh!' + err);
+
+    });
+
+});
+
+
+
+app.get('/categories', (req, res, next) => {
+  // Retrieve a random chuck joke
+  client.getJokeCategories()
+    .then((response) => {
```

Comment:
You can call this response (or whatever you want) but may be confusing because you have a variable called res in the route. Better call it "categories". That is what the API is giving you. Remember to change the variable's name in the line 54 too.